src/coreclr/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrame.NativeAot.cs

```diff
@@ -110,14 +110,26 @@ private void InitializeForIpAddress(IntPtr ipAddress, bool needFileInfo)
         [MethodImplAttribute(MethodImplOptions.NoInlining)]
         private void BuildStackFrame(int frameIndex, bool needFileInfo)
         {
+            IntPtr ipAddress;
+
             const int SystemDiagnosticsStackDepth = 2;
 
-            frameIndex += SystemDiagnosticsStackDepth;
-            IntPtr[] frameArray = new IntPtr[frameIndex + 1];
-            int returnedFrameCount = RuntimeImports.RhGetCurrentThreadStackTrace(frameArray);
-            int realFrameCount = (returnedFrameCount >= 0 ? returnedFrameCount : frameArray.Length);
+            // Unreasonably high or negative frameIndex could lead to overflows or failures to allocate
+            // the frameArray below so spot check it's sane.
+            RuntimeImports.RhGetCurrentThreadStackBounds(out IntPtr stackLow, out IntPtr stackHigh);
```

Comment:
Instead of this, would it be better for `RhGetCurrentThreadStackTrace` to take an integer with how many frames to skip?

Reply:
I reverted this part and block the relevant tests on #103218. This API is really confusing wrt integer overflows and underflows (that we enshrine in tests rather than block) and propagating the confusion to the runtime APIs doesn't sound like an improvement. It should be dealt separately from this PR.